### ISSUE TITLE
Add `getindex`, `setindex!` methods for `::Colon`

### ISSUE THREAD

**Issue by PerezHz** [JuliaDiff/TaylorSeries.jl#100]
On Julia 0.5.1, along with TaylorSeries latest master:

```julia
julia> using TaylorSeries

julia> x = Taylor1(rand(5), 4)
 0.8184698412499662 + 0.6126746629610416 t + 0.7501066040877835 t² + 0.9805243808760169 t³ + 0.8398636369337393 t⁴ + 𝒪(t⁵)

julia> x[1]
0.8184698412499662

julia> x[end]
0.8398636369337393

julia> x[1:end]
5-element Array{Float64,1}:
 0.81847 
 0.612675
 0.750107
 0.980524
 0.839864
```
works fine, but:
```julia
julia> x[:]
ERROR: MethodError: no method matching getindex(::TaylorSeries.Taylor1{Float64}, ::Colon)

julia> x[:] = rand(5)
ERROR: MethodError: no method matching setindex!(::TaylorSeries.Taylor1{Float64}, ::Array{Float64,1}, ::Colon)
```
gives an error. A workaround is to simply use `.coeffs`:
```julia
julia> x.coeffs[:]
5-element Array{Float64,1}:
 0.81847 
 0.612675
 0.750107
 0.980524
 0.839864

julia> x.coeffs[:] = rand(5)
5-element Array{Float64,1}:
 0.0665044
 0.0664276
 0.341507 
 0.646912 
 0.811692 
```
but I think it would be nice to have `getindex` and `setindex!` methods using `::Colon`... what do you guys think @lbenet @dpsanders ?

**Comment by dpsanders** [comment]
Sure, good point -- go ahead and define them!

**Comment by lbenet** [comment]
This has been solved in #101, so I'm closing.
